### ISSUE TITLE
Problem with "Speed on the X axis" action in "Top-down movement" behavior.

### ISSUE THREAD

**Issue by PANDAKO-GitHub** [4ian/GDevelop#5147]
## Describe the bug
When the "Speed on the X axis" action is performed while moving the object in the X direction, the Y axis deceleration appears to be ignored.
This problem occurs even if the X and Y axes are reversed.

## To Reproduce
1. Add a "Top-down movement" behavior to the object and place it in the scene.
2. Create the event as shown in this image.
3. Preview and press the Up or Down key.
4. The object will continue to move even if you release the key.

![image](https://user-images.githubusercontent.com/68310169/227149415-eae0f50f-a033-41ee-8c1b-b7478047e3ab.png)

## Other details
GDevelop 5.1.159
 web-app

**Comment by D8H** [comment]
Yes, it's a bit counter intuitive but it's how this behavior works to avoid the object to slide too much and becoming impossible to control.

Please take a look to this discussion for more details:
- https://github.com/4ian/GDevelop/issues/3802
